### PR TITLE
Fix installation script for containerd on armhf

### LINKED ISSUE #295: containerd fails to install in install.sh on raspberry pi (32bit OS)
## My actions before raising this issue
I'm following serverless for everyone else 1.5, it recommends the use of 32bit Raspberry Pi OS lite

## Why do you need this?
Just trying to follow the guide to setup faasd on a raspberry pi

## Expected Behaviour
arkade should install the cni plugins

## Current Behaviour

```
root@faasd-arm:/srv/faasd# ./hack/install.sh
+ export OWNER=openfaas
+ OWNER=openfaas
+ export REPO=faasd
+ REPO=faasd
+ export ARKADE=/usr/local/bin/arkade
+ ARKADE=/usr/local/bin/arkade
+ '[' -z /root ']'
+ version=
+ echo 'Finding latest version from GitHub'
Finding latest version from GitHub
++ curl -sI https://github.com/openfaas/faasd/releases/latest
++ grep -i location:
++ awk -F/ '{ printf "%s", $NF }'
++ tr -d '\r'
+ version=0.16.2
+ echo 0.16.2
0.16.2
+ '[' '!' 0.16.2 ']'
+ SUDO=sudo
++ id -u
+ '[' 0 -eq 0 ']'
+ SUDO=
+ verify_system
+ '[' -d /run/systemd ']'
+ install_required_packages
++ has_apt_get
+++ command -v apt-get
++ '[' -n /usr/bin/apt-get ']'
+ apt-get update -y
Hit:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease
Hit:2 http://archive.raspberrypi.org/debian bullseye InRelease
Reading package lists... Done
+ apt-get install -y curl runc bridge-utils iptables
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
bridge-utils is already the newest version (1.7-1).
curl is already the newest version (7.74.0-1.3+deb11u2).
iptables is already the newest version (1.8.7-1).
runc is already the newest version (1.0.0~rc93+ds1-5+deb11u2).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+ /sbin/sysctl -w net.ipv4.conf.all.forwarding=1
net.ipv4.conf.all.forwarding = 1
+ echo net.ipv4.conf.all.forwarding=1
+ tee -a /etc/sysctl.conf
net.ipv4.conf.all.forwarding=1
+ install_arkade
+ curl -sLS https://get.arkade.dev
+ sh
armv7l
Downloading package https://github.com/alexellis/arkade/releases/download/0.8.32/arkade-armhf as /tmp/arkade-armhf
Download complete.

Running with sufficient permissions to attempt to move arkade to /usr/local/bin
New version of arkade installed to /usr/local/bin
Creating alias 'ark' for 'arkade'.
            _             _
  __ _ _ __| | ____ _  __| | ___
 / _` | '__| |/ / _` |/ _` |/ _ \
| (_| | |  |   < (_| | (_| |  __/
 \__,_|_|  |_|\_\__,_|\__,_|\___|

Open Source Marketplace For Developer Tools

Version: 0.8.32
Git Commit: b004f9adb0f90c89ef1440f988685fd6b6111d3f

 🐳 arkade needs your support, learn more: https://my.arkade.dev
+ arkade --help
Usage:
  arkade [flags]
  arkade [command]

Available Commands:
  completion  Output shell completion for the given shell (bash or zsh)
  get         The get command downloads a tool
  help        Help about any command
  info        Find info about a Kubernetes app
  install     Install Kubernetes apps from helm charts or YAML files
  kasten      Sponsored Apps for kasten
  system      System apps
  uninstall   Uninstall apps installed with arkade
  update      Print update instructions
  venafi      Sponsored Apps for Venafi
  version     Print the version

Flags:
  -h, --help   help for arkade

Use "arkade [command] --help" for more information about a command.
+ install_cni_plugins
+ cni_version=v0.9.1
+ /usr/local/bin/arkade system install cni --version v0.9.1 --path /opt/cni/bin --progress=false
Installing CNI to /opt/cni/bin
Error: this app only supports x86_64 and aarch64 and not armv7l
```

## Are you a GitHub Sponsor (Yes/No?)

Check at: https://github.com/sponsors/openfaas
- [ ] Yes
- [x] No

## List All Possible Solutions and Workarounds

I could try the 64bit version?

## Steps to Reproduce (for bugs)

1. provisioned a fresh sd card
image
2. ran apt update && apt upgrade
3. cloned in the faasd repo to /srv/faasd (commit 6dbc33d045fd64bbd7669c19933da500e6601acb)
4. ran ./hack/install.sh
5.

## Your Environment

* OS and architecture:
`Raspbian OS Lite`
* Versions:
`11`

go version - not installed

containerd -version - not installed

uname -a
```
Linux faasd-arm 5.15.56-v7+ #1575 SMP Fri Jul 22 20:28:11 BST 2022 armv7l GNU/Linux
```

cat /etc/os-release
```
cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 11 (bullseye)"
NAME="Raspbian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

faasd version - not installed
```

### PR DESCRIPTION
**Pull request by alexellis**
Signed-off-by: Alex Ellis (OpenFaaS Ltd) <alexellis2@gmail.com>

## Description

Fix installation script for containerd on armhf, by downloading from alexellis/containerd-arm as per previous script version - https://github.com/openfaas/faasd/commit/a88997e42c99e0b0d2779c7be99babf75dd550ae

## Motivation and Context

Fixes #295 

## How Has This Been Tested?

I need some help with testing this.

### CODE REVIEW

File: hack/install.sh [hunk 1]
```diff
@@ -84,9 +84,21 @@ install_cni_plugins() {
...
 
 install_containerd() {
-  CONTAINERD_VER=v1.6.4
+
   $SUDO systemctl unmask containerd || :
-  $SUDO $ARKADE system install containerd --systemd --version ${CONTAINERD_VER}  --progress=false
+
+  CONTAINERD_VER=1.6.4
+  arch=$(uname -m)
+  if[ $arch == "armv7l"]; then
```
Comment (welteki) [inline hack/install.sh:92]: Currently the script fails:
```
line 92: syntax error near unexpected token `then'
```

```suggestion
  if [ $arch == "armv7l"]; then
```

**Comment by welteki** [comment]
This solves the issue for installing containerd but the arkade system app for the cni-plugins does not support armv7l arch either.

```
Installing CNI to /opt/cni/bin
Error: this app only supports x86_64 and aarch64 and not armv7l
```

There is a release in the official cni repo so it could be added to arkade.

File: hack/install.sh (hunk 1 above)
Comment (rgee0) [reply hack/install.sh:92]: Concur.  Saw the same.

**Comment by rgee0** [comment]
The CNI thing is interesting.  That hasn't changed here, and how its handled over on Arkade looks unexpected.  

It drops out like we see here before then looking for the `dlArch` and 'downgrades' arm7l to arm6l (of course these paths are never reached because it drops out earlier for these arch's)

https://github.com/alexellis/arkade/blob/f3810b3e6603142566fe88c22d5d55a0d44e759d/cmd/system/cni.go#L53-L63

**Pull request closed without merging**